### ISSUE TITLE
Extra table cell border in Safari

### ISSUE THREAD

**Issue by igordanchenko** [tailwindlabs/tailwindcss#8869]
**What version of Tailwind CSS are you using?**

3.1.6

**What build tool (or framework if it abstracts the build tool) are you using?**

Next.js 12.2.2, postcss 8.4.14, autoprefixer 10.4.7

**What version of Node.js are you using?**

16.15.0

**What browser are you using?**

Safari 15.5

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/EcaFyeEx1T
https://play.tailwindcss.com/1kMB78w9i0

https://codesandbox.io/s/hardcore-gwen-f7rf35?file=/pages/index.js

**Describe your issue**

I'm using `@tailwindcss/typography` plugin and I noticed some erroneous table cell borders appearing in Safari. As far as I can tell, this is only happening in desktop Safari.

Here are some use cases:

1) Increase table font size via `prose-table:text-base` and place `<p>` tags inside `<td>`:

https://play.tailwindcss.com/EcaFyeEx1T

<img width="722" alt="Screen Shot 2022-07-15 at 1 46 10 PM" src="https://user-images.githubusercontent.com/64441155/179282155-910afbf4-69ce-42ee-a9f8-c44721c97b07.png">

2) Place `<ul>` / `<li>` inside `<td>`:

https://play.tailwindcss.com/1kMB78w9i0

<img width="686" alt="Screen Shot 2022-07-15 at 2 09 48 PM" src="https://user-images.githubusercontent.com/64441155/179285184-f04ec3bf-1c24-45e0-ba92-ae9d65f35fc4.png">

It looks like this issue has something to do with elements' margin inside `<td>`'s. But removing the margin just makes this issue less obvious as the extra border is still there, just less visible:

<img width="687" alt="Screen Shot 2022-07-15 at 2 03 27 PM" src="https://user-images.githubusercontent.com/64441155/179284583-306ca04d-d041-469d-96dd-d6abb72bbe6e.png">

**Comment by thecrypticace** [comment]
Leaving a note for later investigation:

Setting `vertical-align` to `top`, `middle`, or `bottom` seems to remove the extra border. The problem is that, in Safari, the table cell is *not* the same height as the adjacent table cells and somehow changing the vertical-align to a non-text-based one seems to prevent or work around the issue.

**Comment by igordanchenko** [comment]
I was able to reproduce this issue outside of Tailwind styling, so it's not a Tailwind issue per se. However it would still be nice to have a workaround in place since `vertical-align: baseline` is the default `<td>` style in Tailwind (and that's what it looks like is causing this issue).

**Comment by thecrypticace** [comment]
I think I'll need to file a browser bug for this. I came up with a handful of passing and failing test cases: https://play.tailwindcss.com/POVW40RJlF

**Comment by thecrypticace** [comment]
Hey, I just filed a WebKit bug for this: https://bugs.webkit.org/show_bug.cgi?id=243167

The only workarounds I have are:
1. Use a border larger than 1px; OR
2. Use border-separate (if that works for your design); OR
3. Only use top/middle/bottom vertical-align

Closing as it's not something we can directly fix. Hopefully it'll get picked up sooner rather than later.

Thanks again for reporting this!
